Alpaca-Finance-Agents/Large-Position-Alert/src/agent.ts

```diff
@@ -0,0 +1,124 @@
+  [VAULT_ADDRESSES[0], BigInt(100000000000000000000000)], // 100,000 BUSD
+  [VAULT_ADDRESSES[1], BigInt(30000000000000000000)],     // 30 ETH
+  [VAULT_ADDRESSES[2], BigInt(3000000000000000000)],      // 3 BTCB
+  [VAULT_ADDRESSES[3], BigInt(100000000000000000000000)], // 100,000 USDT
+  [VAULT_ADDRESSES[4], BigInt(200000000000000000000000)], // 200,000 ALPACA
+  [VAULT_ADDRESSES[5], BigInt(100000000000000000000000)]  // 100,000 TUSD
+])
+
+export const workEventAbi: string = 'event Work(uint256 indexed id, uint256 loan)';
+const VAULT_IFACE: utils.Interface = new utils.Interface([workEventAbi]);
+export const workEventSig: string = VAULT_IFACE.getEvent('Work').format('sighash');
+
+function filterForVault(vaultAddresses: string[], address: string | null): [string, bigint | undefined] {
+  let filteredAddress: string = " ";
+  let filteredThreshold: bigint | undefined = BigInt(0);
+
+  // TODO: CONFIRM USING toLowerCase() IS NOT CAUSING SECONDARY NEGATIVE EFFECTS
+  if(address === vaultAddresses[0].toLowerCase()) {
+    filteredAddress = vaultAddresses[0];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[0])
+  } else if(address === vaultAddresses[1].toLowerCase()) {
+    filteredAddress = vaultAddresses[1];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[1])
+  } else if(address === vaultAddresses[2].toLowerCase()) {
+    filteredAddress = vaultAddresses[2];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[2])
+  } else if(address === vaultAddresses[3].toLowerCase()) {
+    filteredAddress = vaultAddresses[3];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[3])
+  } else if(address === vaultAddresses[4].toLowerCase()) {
+    filteredAddress = vaultAddresses[4];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[4])
+  } else if(address === vaultAddresses[5].toLowerCase()) {
+    filteredAddress = vaultAddresses[5];
+    filteredThreshold = VAULT_MAP.get(vaultAddresses[5])
+  }
+
+  return [filteredAddress, filteredThreshold];
+}
+
+const createFindingGenerator = (alertId: string): FindingGenerator => {
+  return (metadata: { [key: string]: any } | undefined): Finding => {
+    const decodedData = decodeParameters(
+      ["uint256", "uint256"],
+      metadata?.data
+    );
+
+    return Finding.fromObject({
+      name: "Large Position Event",
+      description: "Large Position Has Been Taken",
+      alertId: alertId,
+      severity: FindingSeverity.Info, // TODO: FIND OUT WHICH Severity TO USE HERE (Info?)
+      type: FindingType.Unknown,  // TODO: FIND OUT WHICH Type TO USE HERE (Unknown?)
+      metadata:{
+        positionId: decodedData[0],
+        borrowAmount: decodedData[1]
+      },
+    });
+  };
+};
+
+export function provideHandleTransaction(
+  alertId: string,
+  addresses: string[],
+): HandleTransaction {
+  return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    let [filteredAddress, filteredThreshold] = filterForVault(addresses, txEvent.to);
```

Comment:
why not just `VAULT_ADDRESSES.get(txEvent.to)`? For knowing if it is not there I think you can just check if the result is `undefined`.